yarn-project/noir-compiler/src/cli.ts

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env node
+import nodePath from 'path';
+import fs from 'fs/promises';
+import { Command } from 'commander';
+import { ContractCompiler } from './compile.js';
+
+const program = new Command();
+
+const fileExists = async (path: string) => !!(await fs.stat(path).catch(() => false));
+
+const main = async () => {
+  program
+    .name('aztec_noir_compiler')
+    .command('compile')
+    .argument('[path]', 'Path to the contract project', '.')
+    .action(async (path: string) => {
+      const projectPath = nodePath.resolve(path);
+
+      const compiler = new ContractCompiler(projectPath);
+      const contracts = await compiler.compile();
+
+      const buildFolderPath = nodePath.join(projectPath, 'target');
+      if (!(await fileExists(buildFolderPath))) {
+        await fs.mkdir(buildFolderPath);
+      }
```

Comment:
For next time, [fs-extra](https://www.npmjs.com/package/fs-extra) mkdirp is your friend